likelihood/act-dr6-lens/act_dr6_lenslike_interface.py

```diff
@@ -27,23 +66,34 @@ def execute(block, config):
     # as well as the TT, EE, TE, BB CMB spectra (needed for likelihood corrections)
     # in uK^2 units. All of these are C_ell (not D_ell), no ell or 2pi factors.
     ell = block[names.cmb_cl, 'ell']
+
+    if ell.max() < data_dict['trim_lmax']:
+        raise ValueError(f"An lmax of at least {data_dict['trim_lmax']} is required.")
+
     f1 = ell * (ell + 1) / (2 * np.pi)
     cl_tt = block[names.cmb_cl, 'tt'] / f1
     cl_ee = block[names.cmb_cl, 'ee'] / f1
     cl_te = block[names.cmb_cl, 'te'] / f1
     cl_bb = block[names.cmb_cl, 'bb'] / f1
 
-    # Slightly different normalization here
+    # Slightly different normalization here
     cl_pp = block[names.cmb_cl, 'pp'] / f1
 
-    cl_kk = act_dr6_lenslike.pp_to_kk(cl_pp, ell)
+    if data_dict['varying_cmb_alens']:
+    # if block.has_value(cosmo, 'A_lens'):
+        cl_pp /= block[cosmo, "A_lens"]
+
+    if data_dict['limber']:
+        cl_kk = get_limber_clkk()
+    else:
+        cl_kk = act_dr6_lenslike.pp_to_kk(cl_pp, ell)
 
     # Then call the act code
-    lnlike, bclkk = act_dr6_lenslike.generic_lnlike(data_dict,ell, cl_kk, ell, cl_tt, cl_ee, cl_te, cl_bb)
+    lnlike = act_dr6_lenslike.generic_lnlike(data_dict,ell, cl_kk, ell, cl_tt, cl_ee, cl_te, cl_bb, data_dict['trim_lmax'])
     block[names.likelihoods, 'act_dr6_lens_like'] = lnlike
 
     if not data_dict['cosmosis_like_only']:
-        block[names.data_vector, 'act_dr6_lens_theory'] = bclkk
+        # block[names.data_vector, 'act_dr6_lens_theory'] = bclkk
```

Comment:
It's useful to save the theory, though I know that would require changes to the library.  I'll open an issue there.